### PR TITLE
Add more "Impactful" items to the career ladder.

### PR DESCRIPTION
**Pull request by mathias-abtion**
As mentioned in https://github.com/abtion/guidelines/pull/51, the career ladder is currently being reviewed and updated. This pull request suggests the following titles for the six levels:

* Level 1: Associate Software Engineer
* Level 2: Software Engineer
* Level 3: Software Engineer
* Level 4: Senior Software Engineer
* Level 5: Senior Software Engineer
* Level 6: Principal Software Engineer

In addition, it adds more points under the "Impactful" section of each level to explain the desired behavior of somebody on that particular level. Some of these points should perhaps be moved to "Communication", although communication is, of course, impactful in itself.

### CODE REVIEW

File: career/ladder.md [hunk 1]
```diff
@@ -53,10 +57,20 @@ Here are some examples of other career ladders:
...
+- [ ] Understands and can provide a high-level overview of the current milestone
+  of a project.
+- [ ] Focuses on building strong relationships with their team, direct manager,
+  and project manager/product owner.
+- [ ] Identifies unclear requirements and reaches out to relevant stakeholders
+  to clarify issues.
+- [ ] Reliably do what they say they will, and communicates promptly if an issue
+  arises.
+- [ ] Requires minimal guidance from seniors and can help unblock juniors when
+  necessary.
```
Comment (MarkusAbtion) [inline career/ladder.md:73]: Is this a level 2 or level 3 attribute?

File: career/ladder.md [hunk 2]
```diff
@@ -188,11 +217,14 @@ Here are some examples of other career ladders:
 - [ ] Sought out for technical guidance.
 - [ ] Anticipates technical issues at a product level and works with design to avoid them.
 
-### Level 6
+### Level 6 - Principal Software Engineer
 
 #### Impactful
-- [ ] Has made an obvious positive impact on the entire company's technical trajectory.
-- [ ] Focused on and responsible for their team's productivity and growth.
+- [ ] Plays a key role of the technical direction of the company and has
```
Comment (MarkusAbtion) [inline career/ladder.md:223]:
```suggestion
- [ ] Plays a key role in the technical direction of the company and has
```

... I think

File: career/ladder.md [hunk 3]
```diff
@@ -22,9 +22,13 @@ Here are some examples of other career ladders:
...
 
-### Level 1
+### Level 1 - Associate Software Engineer
 #### Impactful
-- [ ] Contributes positively to the development of one or multiple apps.
+- [ ] Consistently completes work on a task-level within one or multiple apps.
+- [ ] Focuses on building strong relationships with their team and direct
+  manager.
+- [ ] Proactively reaches out for help when struggling with a task.
+- [ ] Actively seeks feedback to improve from colleagues and receives it well.
```
Comment (runephilosof-abtion) [inline career/ladder.md:31]: These sound more like _Communicative_ than **Impactful**

File: career/ladder.md [hunk 4]
```diff
@@ -53,10 +57,20 @@ Here are some examples of other career ladders:
...
 #### Impactful
 - [ ] Capable of providing support for their area, including systems they are not familiar with.
+- [ ] Understands and can provide a high-level overview of the current milestone
+  of a project.
+- [ ] Focuses on building strong relationships with their team, direct manager,
+  and project manager/product owner.
+- [ ] Identifies unclear requirements and reaches out to relevant stakeholders
+  to clarify issues.
+- [ ] Reliably do what they say they will, and communicates promptly if an issue
+  arises.
```
Comment (runephilosof-abtion) [inline career/ladder.md:71]: I would rather put these under _communicative_

File: career/ladder.md [hunk 5]
```diff
@@ -84,11 +98,17 @@ Here are some examples of other career ladders:
...
 
-### Level 3
+### Level 3 - Software Engineer
 
 #### Impactful
 - [ ] Drives tasks from internal meetings and parks them nicely in internal acceptance.
-- [ ] Drives tasks from internal meetings and parks them nicely in internal acceptance.
+- [ ] Understands and can provide a high-level overview of a project.
+- [ ] Consistently facilitates communication with project managers/product
+  owners for one or multiple apps.
```
Comment (runephilosof-abtion) [inline career/ladder.md:107]: Communicative

File: career/ladder.md [hunk 6]
```diff
@@ -119,11 +139,14 @@ Here are some examples of other career ladders:
...
 
-### Level 4
+### Level 4 - Senior Software Engineer
 
 #### Impactful
 - [ ] Partners with project managers and designers to drive technical decisions based on the highest output for all parties' needs.
 - [ ] Drives tasks from client research to client acceptance.
+- [ ] Understands and can provide a detailed overview of a project.
+- [ ] Focuses on building strong relationships with their team, direct
+  manager, and other seniors across the organization.
```
Comment (runephilosof-abtion) [inline career/ladder.md:149]: Communicative

File: career/ladder.md [hunk 7]
```diff
@@ -154,10 +177,16 @@ Here are some examples of other career ladders:
...
+### Level 5 - Senior Software Engineer
 
 #### Impactful
 - [ ] Is recognized as a helpful consultant that consistently delivers high-value solutions for real problems.
+- [ ] Reviews projects critically and ensures that they are divided into parts
+  that can be solved in isolation most of the time.
+- [ ] Consistently shows initiative and identifies areas of improvements for
+  themselves and the team.
+- [ ] Ensures proactively that the relevant stakeholders are informed if the
+  team is running into issues.
```
Comment (runephilosof-abtion) [inline career/ladder.md:189]: Communicative

File: career/ladder.md [hunk 8]
```diff
@@ -188,11 +217,14 @@ Here are some examples of other career ladders:
...
+### Level 6 - Principal Software Engineer
 
 #### Impactful
-- [ ] Has made an obvious positive impact on the entire company's technical trajectory.
-- [ ] Focused on and responsible for their team's productivity and growth.
+- [ ] Plays a key role of the technical direction of the company and has
+  introduced positive changes across multiple teams.
+- [ ] Focuses on the growth of their team's productivity and career.
+- [ ] Provides executives with high-level guidance on technical decisions when
+  necessary.
```
Comment (runephilosof-abtion) [inline career/ladder.md:227]: Communicative

**Review by runephilosof-abtion — CHANGES_REQUESTED** [review]
Nice suggestions.
Some of them I would move to _Communicative_ (the header is called `communication`, but I think it should be renamed (in another PR) :bulb:  ).

File: career/ladder.md (hunk 2 above)
Comment (mathias-abtion) [reply career/ladder.md:223]: Good catch!

**Comment by mathias-abtion** [comment]
@runephilosof: I agree with your suggestions and have now moved the items from "Impactful" to "Communication".

File: career/ladder.md (hunk 1 above)
Comment (mathias-abtion) [reply career/ladder.md]: I think you're right that it should perhaps be moved which I've done now.

**Pull request merged**